src/libPMacc/include/mappings/threads/ThreadCollective.hpp

```diff
@@ -32,72 +32,61 @@
 namespace PMacc
 {
 
-/** execute a functor on a domain
+/** execute a functor for each cell of a domain
  *
- * For each argument passed to the functor the `operator()` is called.
- * This is a **collective** functor and needs to be called by all worker within a block.
+ * the user functor is executed on each elements of the full domain (GUARD +CORE)
  *
- * @tparam T_BlockDomain domain description, type mist contain the definitions `SuperCellSize`,
- *                       `FullSuperCellSize`, `OffsetOrigin` and `Dim`
- * @tparam T_numWorkers number of workers which are used to execute this functor
+ * @tparam T_DataDomain PMacc::SuperCellDescription, compile time data domain
+ *                      description with a CORE and GUARD
+ * @tparam T_numWorkers number of workers
  */
 template<
-    typename T_BlockDomain,
-    uint32_t T_numWorkers = math::CT::volume<typename T_BlockDomain::SuperCellSize>::type::value
+    typename T_DataDomain,
+    uint32_t T_numWorkers
```

Comment:
I removed that because it is mixing the data domain and the worker domain and creates a one to one relation between both. 

SHort: this is not compatible with the lock step programming model